pdpserver/server/ctrl_storage.go

```diff
@@ -0,0 +1,49 @@
+package server
+
+import (
+	"io"
+	"net/http"
+	"strconv"
+	"strings"
+)
+
+type storageHandler struct {
+	s *Server
+}
+
+func (handler *storageHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	// sanity check
+	root := handler.s.p
+	if root == nil {
+		io.WriteString(w, "Server missing policy storage")
+		return
+	}
+
+	// parse depth
+	queryOpt := r.URL.Query()
+	depthOpt, ok := queryOpt["depth"]
+	if !ok {
+		io.WriteString(w, "depth option not specified")
+		return
+	}
+	depthStr := depthOpt[0]
+	depth, err := strconv.ParseInt(depthStr, 10, 64)
+	if err != nil {
+		io.WriteString(w, err.Error())
+		return
+	}
+
+	// parse path
+	path := strings.FieldsFunc(r.URL.Path, func(c rune) bool { return c == '/' })[1:]
```

Comment:
Why not `strings.Split(r.URL.Path, "/")`?

Reply:
I wasn't sure if http would properly parse out dangling '/'. Example: "127.0.0.1:8080/query/abc/def/?depth=12" => path = "query/abc/def/" => ["query", "abc", "def", ""]. FieldsFunc drops empty strings

Reply:
On second thought, I'll just use Split, since dangling / can just be a 400 error

Reply:
Ok!